src/scanner.c

```diff
@@ -0,0 +1,62 @@
+#include <tree_sitter/parser.h>
+
+enum {
+  COMMENT
+};
+
+void *tree_sitter_ocaml_external_scanner_create() {
+  return malloc(sizeof(int));
+}
+
+void tree_sitter_ocaml_external_scanner_destroy(void *payload) {
+  free(payload);
+}
+
+void tree_sitter_external_extra_tokens_external_scanner_reset(void *payload) {}
+
+unsigned tree_sitter_ocaml_external_scanner_serialize(void *payload, char *buffer) {
+  return true;
+}
+
+void tree_sitter_ocaml_external_scanner_deserialize(void *payload, const char *buffer, unsigned length) {}
+
+bool tree_sitter_ocaml_external_scanner_scan(void *payload, TSLexer *lexer, const bool *whitelist) {
+  int *depth = (int*) payload;
```

Comment:
Could this just be `int depth = 0;`? I would think we wouldn't need to carry any state between calls to `scan`.

Reply:
That's right, I was still trying to figure out how external works when I wrote this.